### PR TITLE
Created post request for streaks

### LINKED ISSUE #48: POST request for streaks
Add a row to the database every day the site is visited. Make sure there are no duplicates. An example POST request can be found [here](https://github.com/cse110-sp24-group14/test-mysql) with `addUser`. 

[Database documentation](https://github.com/cse110-sp24-group14/cse110-sp24-group14/blob/main/specs/adr/052124-db-schema.md)

Use `SiteVisits` table

### PR DESCRIPTION
**Pull request by hritikbharucha**
## Tracking Info

Resolves #48 

## Changes

- Set a date variable to today's date and formatted it
- Created a SQL query for the date to check if it already exists
- Created another SQL query for the date to insert it if it doesn't exist

## Testing

- I'm not sure how to test this because my task only posts the request and does not get any data to display.

## Confirmation of Change

The task for getting the streaks should be able to properly get the data from `siteVisits` so that should confirm that the data was correctly posted.

### CODE REVIEW

File: source/server.js [hunk 1]
```diff
@@ -16,6 +16,33 @@ const connection = mysql.createConnection({
...
 });
 
+const addStreaks = (callback) => {
+    const today = new Date();
+    today.setHours(0, 0, 0, 0);  // Set to the start of the day
+    const formattedDate = today.toISOString().split('T')[0];  // Format date as YYYY-MM-DD
+
+    // First, check if there's already a visit for today
+    const checkQuery = 'SELECT * FROM SiteVisits WHERE visit_date = ?';
+    connection.query(checkQuery, [formattedDate], (checkError, checkResults) => {
```
Comment (dylantrann) [inline source/server.js:26]: what does `[formattedDate]` do? i tried looking it up but couldn't find anything describing it. if it just replaces the `?` from `checkQuery` with the date then it should be all good

File: source/server.js [hunk 2]
```diff
@@ -16,6 +16,33 @@ const connection = mysql.createConnection({
...
+        if (checkError) {
+            callback(checkError);
+        } else if (checkResults.length > 0) {
+            // If a record for today exists, do nothing (or handle as needed)
+            callback(null, { message: 'Visit already recorded for today' });
+        } else {
+            // If no record for today exists, insert a new record
+            const insertQuery = 'INSERT INTO SiteVisits (visit_date) VALUES (?)';
+            connection.query(insertQuery, [formattedDate], (insertError, insertResults) => {
```
Comment (dylantrann) [inline source/server.js:35]: same as above comment

File: source/server.js (hunk 1 above)
Comment (hritikbharucha) [reply source/server.js:26]: It's just the query we search for similar to how the addUser function creates a post request for the `user.username` and `user.email` in the test-mysql repository [here](https://github.com/cse110-sp24-group14/test-mysql/blob/main/server.js).

Comment (dylantrann) [reply source/server.js:26]: make sure that `[formattedDate]` replaces `?` as so: `'2024-5-20'` 

without the apostrophes, it won't function correctly

Comment (hritikbharucha) [reply source/server.js:26]: Yeah it's defined as `const formattedDate = today.toISOString().split('T')[0]; ` and when I test with `oftype formattedDate` it returns `string` so would that be the right format?

**Review by dylantrann — COMMENTED** [review]
I verified that the checkQuery calls work as intended by testing with Docker. I also verified that the formattedDate is correct using node in terminal.

I would get another person's perspective on this review since I'm not sure how to test the database functionalities in regards to the JavaScript.

**Review by brentonmdunn — APPROVED** [review]

**Pull request merged**